### PR TITLE
Gcw 2216 add dashboard element to homepage

### PR DESCRIPTION
**Pull request by abulsayyad123**
Hi Team,
This is PR for JIRA ticket 2216- Add dashboard element to stock. https://jira.crossroads.org.hk/browse/GCW-2216.
This PR consists of :
- Orders count of different type i.e submitted, processing, awaiting_dispatch, schedule and priority.
- Clicking on the list redirects to orders page with filter applied and highlighting the selected filter.
- Searching is now working according to selected filtered. Have configured it with API (GCW-2193).
- Test case (and one failing test case from login).

Please review the PR and let me know if any change is required.

### CODE REVIEW

File: app/controllers/index.js [hunk 1]
```diff
@@ -11,6 +12,17 @@ export default Ember.Controller.extend({
...
     logMeOut() {
       this.get('application').send('logMeOut');
+    },
+
+    goToOrder(states, priority= false) {
+      let stateFilter = [states];
+      if (priority) {
+        stateFilter.unshift('showPriority');
+      }
+      this.get('filterService').clearFilters();
```
Comment (Darshan-Chauhan) [inline app/controllers/index.js]: `this.get('filterService')` can be cached

File: app/controllers/search_module.js [hunk 2]
```diff
@@ -70,7 +72,14 @@ export default Ember.Controller.extend(InfinityRoute, {
       this.set("hasNoResults", false);
       if(this.get("unloadAll")) { this.get("store").unloadAll(); }
 
-      const paginationOpts = { perPage: 25, startingPage: 1, modelPath: 'filteredResults', stockRequest: true };
+      let filter = this.get('filterService').getOrderStateFilters();
```
Comment (Darshan-Chauhan) [inline app/controllers/search_module.js]: ^

File: app/controllers/search_module.js [hunk 3]
```diff
@@ -70,7 +72,14 @@ export default Ember.Controller.extend(InfinityRoute, {
...
+
+      let isPriority = this.get('filterService').isPriority();
+      if (isPriority) {
+        filter.shift();
+      }
+      let typesFilter = this.get('filterService').getOrderTypeFilters();
```
Comment (Darshan-Chauhan) [inline app/controllers/search_module.js]: ^

File: app/controllers/search_module.js [hunk 4]
```diff
@@ -90,6 +99,10 @@ export default Ember.Controller.extend(InfinityRoute, {
     this.set("filteredResults", []);
   },
 
+  stringifyArray(stateOrType) {
+    return this.get("utilityMethods").arrayExists(stateOrType) ? stateOrType.toString() : '';
```
Comment (Darshan-Chauhan) [inline app/controllers/search_module.js]: I think using `Array.isArray(stateOrType)` would be better choice. What do you think?

File: app/services/session.js [hunk 5]
```diff
@@ -16,6 +16,13 @@ export default Ember.Service.extend({
     return store.peekAll('user_profile').get('firstObject') || null;
   }).volatile(),
 
+  isOrderFulfilmentUser: Ember.computed('isLoggedIn', function() {
```
Comment (Darshan-Chauhan) [inline app/services/session.js]: Move this to model... `hasRole` property would be better choice in model. 
Session shouldn't be used to check if User has role or not

File: app/routes/index.js [hunk 6]
```diff
@@ -1,7 +1,11 @@
+import Ember from 'ember';
 import SessionRoute from './session';
+import AjaxPromise from 'stock/utils/ajax-promise';
 
 export default SessionRoute.extend({
   model() {
+    let isOrderFulfilmentRole = this.get('session.isOrderFulfilmentUser');
```
Comment (Darshan-Chauhan) [inline app/routes/index.js]: `session.currentUser.hasRole("order fulfilment")` would be better?

File: app/routes/orders/index.js [hunk 7]
```diff
@@ -1,13 +1,44 @@
...
+import Ember from "ember";
 import AjaxPromise from 'stock/utils/ajax-promise';
 
 export default AuthorizeRoute.extend({
-  model() {
+  filterService: Ember.inject.service(),
+  utilityMethods: Ember.inject.service(),
+
+  model(params) {
+    let filter = this.get('filterService').getOrderStateFilters();
```
Comment (Darshan-Chauhan) [inline app/routes/orders/index.js]: `this.get('filterService')` can be cached

File: app/routes/orders/index.js [hunk 8]
```diff
@@ -1,13 +1,44 @@
...
+
     if(!this.session.get("currentUser")) {
-      return new AjaxPromise("/auth/current_user_profile", "GET", this.session.get("authToken"))
+      new AjaxPromise("/auth/current_user_profile", "GET", this.session.get("authToken"))
         .then(data => {
           this.store.pushPayload(data);
         });
     }
+
+    if (this.get("utilityMethods").arrayExists(filter) && params.isFiltered) {
```
Comment (Darshan-Chauhan) [inline app/routes/orders/index.js]: `Array.isArray()` would be a better choice

File: app/services/filter-service.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+      return this.getOrderStateFilters().indexOf('showPriority') >= 0;
+    }
+  },
+
+  setStateTypeFilter(states) {
+    window.localStorage.setItem('orderStateFilters', JSON.stringify(states));
+  },
+
+  clearFilters() {
+    window.localStorage.setItem('orderStateFilters', []);
```
Comment (Darshan-Chauhan) [inline app/services/filter-service.js]: Should this clear orderType filters also?

File: app/controllers/search_module.js [hunk 10]
```diff
@@ -70,7 +72,14 @@ export default Ember.Controller.extend(InfinityRoute, {
...
 
-      const paginationOpts = { perPage: 25, startingPage: 1, modelPath: 'filteredResults', stockRequest: true };
+      let filter = this.get('filterService').getOrderStateFilters();
+
+      let isPriority = this.get('filterService').isPriority();
+      if (isPriority) {
+        filter.shift();
+      }
+      let typesFilter = this.get('filterService').getOrderTypeFilters();
+      const paginationOpts = { perPage: 25, startingPage: 1, modelPath: 'filteredResults', stockRequest: true, state: this.stringifyArray(filter), type: this.stringifyArray(typesFilter), priority: isPriority };
```
Comment (abhinav098) [inline app/controllers/search_module.js]: can we have 80 chars per line ?

**Pull request merged**